std/range/package.d

```diff
@@ -1434,6 +1434,36 @@ if (isInputRange!(Unqual!R1) && isInputRange!(Unqual!R2) &&
     assert(result.equal(only(6, 7, 8, 9)));
 }
 
+//https://issues.dlang.org/show_bug.cgi?id=19738
+@safe nothrow pure @nogc unittest
+{
+    struct EvilRange
+    {
+        enum empty = true;
+        int front;
+        void popFront() @safe {}
+        auto opAssign(const ref EvilRange other)
+        {
+            *(cast(uint*)0xcafebabe) = 0xdeadbeef;
+            return this;
+        }
+    }
+    enum isDIP1000 = __traits(compiles, () @safe
+    {
+      int x;
+      int* p;
+      p = &x;
+    });
+    static if (isDIP1000)
```

Comment:
Huh? This should never compile; DIP1000 or not.

Reply:
Fixed.